src/prognostic_equations/advection.jl

```diff
@@ -422,6 +427,284 @@ function edmfx_sgs_vertical_advection_tendency!(
                 edmfx_upwinding,
             )
             @. Yₜ.c.sgsʲs.:($$j).q_sno += va
+
+            FT = eltype(params)
+            (; ᶜwₗʲs, ᶜwᵢʲs, ᶜwᵣʲs, ᶜwₛʲs, ᶜtsʲs) = p.precomputed
+
+            ᶠwₗ³ʲs =
+                (@. lazy(CT3(ᶠinterp(Geometry.WVector(-1 * ᶜwₗʲs.:($$j))))))
+            ᶠwᵢ³ʲs =
+                (@. lazy(CT3(ᶠinterp(Geometry.WVector(-1 * ᶜwᵢʲs.:($$j))))))
+            ᶠwᵣ³ʲs =
+                (@. lazy(CT3(ᶠinterp(Geometry.WVector(-1 * ᶜwᵣʲs.:($$j))))))
+            ᶠwₛ³ʲs =
+                (@. lazy(CT3(ᶠinterp(Geometry.WVector(-1 * ᶜwₛʲs.:($$j))))))
```

Comment:
I think it would be a bit confusing if we need to remember ᶠwₗ³ʲs and ᶜwₗʲs have different signs but I'm fine with that.

Reply:
Could you remove the `s` in the variable name though, as it is for one updraft?

Reply:
Agreed on the sign confusion. I'm hoping to soon make it lazy. Then we will not even need to store `ᶜwₗʲs` and we will directly go to the `CT3(ᶠinterp(Geometry.WVector(-1 *` version via `lazy`